### PR TITLE
Proxy for Account abstraction microservice

### PR DESCRIPTION
**Pull request by vbaranov**
Resolves https://github.com/blockscout/blockscout/issues/9108

## Motivation

Create proxy for Account abstraction microservice

## Changelog

Corresponding account abstraction service API endpoints proxied to these:
```
GET /api/v2/account-abstraction/operations/:operation_hash_param
GET /api/v2/account-abstraction/bundlers/:address_hash_param
GET /api/v2/account-abstraction/bundlers
GET /api/v2/account-abstraction/factories/:address_hash_param
GET /api/v2/account-abstraction/factories
GET /api/v2/account-abstraction/paymasters/:address_hash_param
GET /api/v2/account-abstraction/paymasters
GET /api/v2/account-abstraction/accounts/:address_hash_param
GET /api/v2/account-abstraction/accounts
GET /api/v2/account-abstraction/bundles
GET /api/v2/account-abstraction/operations
```
query params are preserved as is.
Address output params are extended to https://github.com/blockscout/blockscout/blob/502dec051c88e98a3a5beffb5e93be43a2957418/apps/block_scout_web/lib/block_scout_web/views/api/v2/helper.ex#L51-L60

## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [x] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [x] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. https://github.com/blockscout/docs/pull/222
  - [x] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex [hunk 1]
```diff
@@ -0,0 +1,180 @@
+defmodule BlockScoutWeb.API.V2.Proxy.AccountAbstractionController do
```
Comment (nikitosing) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex:1]: Unhandled `{:error, :disabled}` response from AccountAbstraction module:
```
2024-01-14T12:56:58.258 [error] #PID<0.2751.0> running BlockScoutWeb.Endpoint (connection #PID<0.2750.0>, stream id 1) terminated
Server: localhost:4000 (http)
Request: GET /api/v2/proxy/account-abstraction/bundlers
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Plug.Conn.Status.code/1
        (plug 1.15.2) lib/plug/conn/status.ex:114: Plug.Conn.Status.code(:error)
        (plug 1.15.2) lib/plug/conn.ex:411: Plug.Conn.put_status/2
        (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex:45: BlockScoutWeb.API.V2.Proxy.AccountAbstractionController.bundlers/2
        (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex:1: BlockScoutWeb.API.V2.Proxy.AccountAbstractionController.action/2
        (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex:1: BlockScoutWeb.API.V2.Proxy.AccountAbstractionController.phoenix_controller_pipeline/2
        (phoenix 1.5.14) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
        (phoenix 1.5.14) lib/phoenix/router/route.ex:41: Phoenix.Router.Route.call/2
        (phoenix 1.5.14) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
        (block_scout_web 6.0.0) lib/block_scout_web/endpoint.ex:1: BlockScoutWeb.Endpoint.plug_builder_call/2
        (block_scout_web 6.0.0) /Users/nikitosing/work/blockscout/blockscout/deps/plug/lib/plug/debugger.ex:136: BlockScoutWeb.Endpoint."call (overridable 3)"/2
        (block_scout_web 6.0.0) lib/block_scout_web/endpoint.ex:1: BlockScoutWeb.Endpoint."call (overridable 4)"/2
        (block_scout_web 6.0.0) /Users/nikitosing/work/blockscout/blockscout/deps/spandex_phoenix/lib/spandex_phoenix.ex:151: BlockScoutWeb.Endpoint.call/2
        (phoenix 1.5.14) lib/phoenix/endpoint/cowboy2_handler.ex:65: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy 2.10.0) /Users/nikitosing/work/blockscout/blockscout/deps/cowboy/src/cowboy_handler.erl:37: :cowboy_handler.execute/2
        (cowboy 2.10.0) /Users/nikitosing/work/blockscout/blockscout/deps/cowboy/src/cowboy_stream_h.erl:306: :cowboy_stream_h.execute/3
        (cowboy 2.10.0) /Users/nikitosing/work/blockscout/blockscout/deps/cowboy/src/cowboy_stream_h.erl:295: :cowboy_stream_h.request_process/3
        (stdlib 4.3.1.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
```
```suggestion
defmodule BlockScoutWeb.API.V2.Proxy.AccountAbstractionController do
```

File: apps/block_scout_web/lib/block_scout_web/api_router.ex [hunk 2]
```diff
@@ -307,6 +307,20 @@ defmodule BlockScoutWeb.ApiRouter do
...
+        get("/bundlers/:address_hash_param", V2.Proxy.AccountAbstractionController, :bundler)
+        get("/bundlers", V2.Proxy.AccountAbstractionController, :bundlers)
+        get("/factories/:address_hash_param", V2.Proxy.AccountAbstractionController, :factory)
+        get("/factories", V2.Proxy.AccountAbstractionController, :factories)
+        get("/paymasters/:address_hash_param", V2.Proxy.AccountAbstractionController, :paymaster)
+        get("/paymasters", V2.Proxy.AccountAbstractionController, :paymasters)
+        get("/accounts/:address_hash_param", V2.Proxy.AccountAbstractionController, :account)
+        get("/accounts", V2.Proxy.AccountAbstractionController, :accounts)
+        get("/bundles", V2.Proxy.AccountAbstractionController, :bundles)
+        get("/operations", V2.Proxy.AccountAbstractionController, :operations)
```
Comment (nikitosing) [inline apps/block_scout_web/lib/block_scout_web/api_router.ex:322]:
```
2024-01-14T13:16:45.475 application=explorer request_id=F6ovSlvSLmAAlccAAARj [error] Error while sending request to Account Abstraction microservice url: ***/api/v1/operations: %HTTPoison.Response{status_code: 404, body: "", headers: [{"Date", "Sun, 14 Jan 2024 10:16:45 GMT"}, {"Content-Length", "0"}, {"Connection", "keep-alive"}, {"strict-transport-security", "max-age=31536000; includeSubDomains"}, {"CF-Cache-Status", "DYNAMIC"}, {"Report-To", "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=gzcuSAj2iGvHFnYi7Sqd4CwSvX8SOGI6AknLtV31eNUxbsNWElpMXaxKkBB4ID0LrUHF0ry2vzG3c2ROnBS3m8ckfqDgG66FPxw24QfRvUlvsYkmacnV13L%2FhfsOWeLkkcHGVshmF2sQ8iIYLlwuru0O%2BU7lL3jybQaCCtw0twg%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}"}, {"NEL", "{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}"}, {"Server", "cloudflare"}, {"CF-RAY", "8455201389af1628-DME"}, {"alt-svc", "h3=\":443\"; ma=86400"}], request_url: "***/api/v1/operations", request: %HTTPoison.Request{method: :get, url: "***/api/v1/operations", headers: [], body: "", params: %{}, options: [params: %{}]}}
2024-01-14T13:16:45.497 application=api_v2 request_id=F6ovSlvSLmAAlccAAARj [info] Sent 500 in 167ms on GET /api/v2/proxy/account-abstraction/operations?
2024-01-14T13:16:45.498 [error] #PID<0.1882.0> running BlockScoutWeb.Endpoint (connection #PID<0.1848.0>, stream id 3) terminated
Server: localhost:4000 (http)
Request: GET /api/v2/proxy/account-abstraction/operations
** (exit) an exception was raised:
    ** (MatchError) no match of right hand side value: {:error, %Jason.DecodeError{position: 0, token: nil, data: ""}}
        (explorer 6.0.0) lib/explorer/microservice_interfaces/account_abstraction.ex:148: Explorer.MicroserviceInterfaces.AccountAbstraction.http_get_request/2
        (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex:138: BlockScoutWeb.API.V2.Proxy.AccountAbstractionController.operations/2
        (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex:1: BlockScoutWeb.API.V2.Proxy.AccountAbstractionController.action/2
        (block_scout_web 6.0.0) lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex:1: BlockScoutWeb.API.V2.Proxy.AccountAbstractionController.phoenix_controller_pipeline/2
        (phoenix 1.5.14) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
        (phoenix 1.5.14) lib/phoenix/router/route.ex:41: Phoenix.Router.Route.call/2
        (phoenix 1.5.14) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
        (block_scout_web 6.0.0) lib/block_scout_web/endpoint.ex:1: BlockScoutWeb.Endpoint.plug_builder_call/2
        (block_scout_web 6.0.0) /Users/nikitosing/work/blockscout/blockscout/deps/plug/lib/plug/debugger.ex:136: BlockScoutWeb.Endpoint."call (overridable 3)"/2
        (block_scout_web 6.0.0) lib/block_scout_web/endpoint.ex:1: BlockScoutWeb.Endpoint."call (overridable 4)"/2
        (block_scout_web 6.0.0) /Users/nikitosing/work/blockscout/blockscout/deps/spandex_phoenix/lib/spandex_phoenix.ex:151: BlockScoutWeb.Endpoint.call/2
        (phoenix 1.5.14) lib/phoenix/endpoint/cowboy2_handler.ex:65: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy 2.10.0) /Users/nikitosing/work/blockscout/blockscout/deps/cowboy/src/cowboy_handler.erl:37: :cowboy_handler.execute/2
        (cowboy 2.10.0) /Users/nikitosing/work/blockscout/blockscout/deps/cowboy/src/cowboy_stream_h.erl:306: :cowboy_stream_h.execute/3
        (cowboy 2.10.0) /Users/nikitosing/work/blockscout/blockscout/deps/cowboy/src/cowboy_stream_h.erl:295: :cowboy_stream_h.request_process/3
        (stdlib 4.3.1.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
```

File: apps/explorer/lib/explorer/microservice_interfaces/account_abstraction.ex [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  """
+
+  alias Explorer.Utility.Microservice
+  alias HTTPoison.Response
+  require Logger
+
+  @doc """
+    Get user operation by hash via GET {{baseUrl}}/api/v1/userOps/:hash
+  """
+  @spec get_user_ops_by_hash(binary()) :: {non_neg_integer(), map()}
```
[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/explorer/lib/explorer/microservice_interfaces/account_abstraction.ex]: Fixed in https://github.com/blockscout/blockscout/pull/9145/commits/56202e0c397e8f3eeeb6eba0010e73d76f213d25

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex (hunk 1 above)
Comment (vbaranov) [reply apps/block_scout_web/lib/block_scout_web/controllers/api/v2/proxy/account_abstraction_controller.ex:1]: Fixed in https://github.com/blockscout/blockscout/pull/9145/commits/56202e0c397e8f3eeeb6eba0010e73d76f213d25

File: apps/block_scout_web/lib/block_scout_web/api_router.ex (hunk 2 above)
Comment (vbaranov) [reply apps/block_scout_web/lib/block_scout_web/api_router.ex:322]: Fixed in https://github.com/blockscout/blockscout/pull/9145/commits/56202e0c397e8f3eeeb6eba0010e73d76f213d25

File: apps/explorer/lib/explorer/microservice_interfaces/account_abstraction.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (nikitosing) [reply apps/explorer/lib/explorer/microservice_interfaces/account_abstraction.ex]: @vbaranov other functions in this module also can return {:error, :disabled}, lets update spec for them too

[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/explorer/lib/explorer/microservice_interfaces/account_abstraction.ex]: Sure, fixed here https://github.com/blockscout/blockscout/pull/9145/commits/9fce1c3ebfc7388a9da17e1ba1efc760c530e326

**Review by nikitosing — APPROVED** [review]

**Review by Qwerty5Uiop — APPROVED** [review]

**Review by varasev — APPROVED** [review]

**Pull request merged**